io.github.sxyazi.yazi.yml

```diff
@@ -0,0 +1,170 @@
+id: io.github.sxyazi.yazi
+runtime: org.freedesktop.Platform
+runtime-version: &runtime-version '24.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+  - org.freedesktop.Sdk.Extension.golang
+command: yazi
+add-extensions:
+  # Needs ffmpeg to generate video preview thumbnails
+  org.freedesktop.Platform.ffmpeg-full:
+    directory: lib/ffmpeg
+    add-ld-path: .
+    version: *runtime-version
+    no-autodownload: false
+    autodelete: true
+
+finish-args:
+  # For downloading plugins
+  - --share=network
+  # Use host-spawn to call utilities not packaged in the sandbox
+  - --talk-name=org.freedesktop.Flatpak
```

Comment:
The package manager (`ya`) needs networking access to download plugins and themes. This is why `git` got packaged as well. 

For `host-spawn`, the main use case is to launch editors on the host (e.g. neovim). There are some other situations such as plugins requiring additional commands to function. 

If any of them is too much permission to grant, I can disable it by default and add a note about it in the README in case someone needs to use `ya` or launch a program outside the sandbox.

Reply:
I think, not allowing to launch external editors is fine considering some other things don't work with the Flatpak either. People who want it can manually add it.

> There are some other situations such as plugins requiring additional commands to function.

What are these generally? Can these be included in the manifest?

Reply:
I removed both permissions since they are needed only in specific situations. I expanded the README file with instructions on how to enable network and `host-spawn` access when needed.

There is a list of plugins [here](https://yazi-rs.github.io/docs/resources/). A lot of them expand the preview feature with a library, so it would be impractical to package every single one of them. These libraries are also subject to change as more plugins appear. I think packaging whatever the base program needs is sufficient. Plugin users can use workarounds.